pyrobosim/pyrobosim/manipulation/grasping.py

```diff
@@ -0,0 +1,547 @@
+        
+        :param object_pose: The pose of the object center, defaults to identity transform
+        :type object_pose: :class:`pyrobosim.utils.pose.Pose`, optional
+        :param robot_pose: The pose of the robot. If none specified, it is not used in calculations.
+        :type robot_pose: :class:`pyrobosim.utils.pose.Pose`, optional
+        :return: Rotation matrix transforming the nominal cuboid orientation to a robot-facing orientation
+        :rtype: :class:`numpy.ndarray`
+        """
+        rot_matrix = np.eye(3)
+        if robot_pose is None:
+            return rot_matrix
+
+        tform_robot_to_object = np.matmul(
+            np.linalg.inv(robot_pose.get_transform_matrix()),
+            object_pose.get_transform_matrix())
+        v_robot_to_object = np.dot(
+            tform_robot_to_object[:3, :3].T, tform_robot_to_object[:3, 3])
+
+        # Figure out what the front face is
+        v_robot_rt_object_proj = np.array([-v_robot_to_object[0],
+                                           -v_robot_to_object[1],
+                                           0.0])
+        v_robot_rt_object_proj /= np.linalg.norm(v_robot_rt_object_proj)
+        all_grasp_faces = [GraspFace.FRONT, GraspFace.BACK,
+                           GraspFace.TOP, GraspFace.BOTTOM,
+                           GraspFace.LEFT, GraspFace.RIGHT]
+        max_dot_prod = -10  # Unrealistic value for dot product
+        for face in all_grasp_faces:
+            dot_prod = np.dot(v_robot_rt_object_proj, normal_from_face[face])
+            if dot_prod > max_dot_prod:
+                max_dot_prod = dot_prod
+                front_face_dir = face
+
+        # Figure out what the top face is
+        unit_z = np.dot(
+            tform_robot_to_object[:3, :3].T,
+            np.array([[0.0], [0.0], [1.0]])).T
+        max_dot_prod = -10  # Unrealistic value for dot product
+        for face in all_grasp_faces:
+            dot_prod = np.dot(unit_z, normal_from_face[face])
+            if dot_prod > max_dot_prod:
+                max_dot_prod = dot_prod
+                top_face_dir = face
+
+        # Compute the transform from nominal coordinates to robot-facing
+        x_vec = -1.0 * normal_from_face[front_face_dir]
+        z_vec = normal_from_face[top_face_dir]
+        y_vec = np.cross(-1.0 * x_vec, z_vec)
+        rot_matrix[:,0] = x_vec
+        rot_matrix[:,1] = y_vec
+        rot_matrix[:,2] = z_vec
+        return rot_matrix
+
+    def should_try_grasp(self, faces_enabled, face_normals, face_vec):
+        """
+        Helper function to validate whether to compute grasps on a specific face.
+        
+        :param faces_enabled: Faces enabled, in the form (front, top, side)
+        :type faces_enabled: list[bool]
+        :param face_normals: Face normals in the canonical directions, in the form (front, top, left, right)
+        :type face_normals: list[:class:`numpy.ndarray`]
+        :param face_vec: Face vector to check against
```

Comment:
Can you explain what this is any better? Like, is it the face that's normal to the robot? In other words, the robot can grasp `face_vec` easily, so let's check if the actual cube faces are close to that?

Reply:
Yeah, this is supposed to be the actual face. What we want to ask in this function, in English, is:

"Given the types of grasps I want to allow, and some map that tells me which face is which from the perspective of the robot, should I generate a grasp in e.g. the "+X" direction?"

and `face_vec` specifically is that last part. I'll try clarify here too

Reply:
Added more documentation!